plugins/module_utils/prerequisites.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+
+# (c) 2023, Jan Kodera (jkodera@purestorage.com)
+# GNU General Public License v3.0+ (see COPYING.GPLv3 or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+import re
+import importlib
+import importlib.metadata
+
+# This file exists because Ansible currently cannot declare dependencies on Python modules.
+# see https://github.com/ansible/ansible/issues/62733 for more info about lack of req support
+
+#############################
+
+# 'module_name, package_name, version_requirements' triplets
+DEPENDENCIES = [
+    ("fusion", "purefusion", ">=1.0.11,<2.0"),
+    ("urllib3", "urllib3", None),
+]
+
+#############################
+
+
+# returns tuple (MAJOR, MINOR, PATCH)
+def _parse_version(val):
+    # regexes for this were really ugly
```

Comment:
As we don't have typing, let's add some docstrings for future generations... :) While the comment above the function is visible only if you open the file, the docstring are usually supported by IDEs' helpers. The same for other functions.
```suggestion
def _parse_version(val):
    """ Takes a string representation of version, returns tuple of ints (MAJOR, MINOR, PATCH) or None if not parsed."""
    # regexes for this were really ugly
```
or some other docstring style...
```suggestion
def _parse_version(val):
    """
    Parse a package version.
    
    :param val: a string representation of the package version
    :returns: tuple of ints (MAJOR, MINOR, PATCH) or None if not parsed
    """
    # regexes for this were really ugly
```